### PR TITLE
Add facets API endpoint

### PR DESCRIPTION
**Pull request by andrewm89**

### CODE REVIEW

File: invenio_communities/facets.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                urlkwargs.add(name, v)
+
+    return (filters, urlkwargs)
+
+
+def _query_filter(search, urlkwargs, definitions):
+    """Ingest query filter in query."""
+    filters, urlkwargs = _create_filter_dsl(urlkwargs, definitions)
+
+    for filter_ in filters:
```
Comment (blixhavn) [inline invenio_communities/facets.py:58]: Good naming :)

File: invenio_communities/links.py [hunk 2]
```diff
@@ -61,3 +61,17 @@ def default_links_pagination_factory(page, urlkwargs):
                                 **urlkwargs)
 
     return links
+
+def default_links_facets_factory():
```
Comment (blixhavn) [inline invenio_communities/links.py:65]: This function is defined here, but not called in this PR. Is it for later use?

**Comment by blixhavn** [comment]
One small question, but :shipit:

**Comment by audub** [comment]
For me, it looks good.

File: invenio_communities/links.py (hunk 2 above)
Comment (andrewm89) [reply invenio_communities/links.py:65]: @blixhavn it is used with the endpoint definition in the api.py file inside view_func.

File: invenio_communities/facets.py (hunk 1 above)
Comment (greut) [reply invenio_communities/facets.py:58]:
``` python
search = functools.reduce(lambda s, f: s.filter(f), filters, search)
```

**Pull request merged**